lib/runtime/wasmer/instance.go

```diff
@@ -216,6 +216,50 @@ func (in *Instance) UpdateRuntimeCode(code []byte) error {
 	return nil
 }
 
+// CheckRuntimeVersion calculates runtime Version for runtime blob passed in
+func (in *Instance) CheckRuntimeVersion(code []byte) (runtime.Version, error) {
```

Comment:
this function has code that's mostly the same as `UpdateRuntimeVersion`, could you put the code that overlaps into a separate function and call that in both?

Reply:
created `setupInstanceVM` function to handle redundant code.